docs/Microsoft.Omex.Gating/GatingTutorial.md

```diff
@@ -0,0 +1,106 @@
+# How to use the Gating library
+
+In your code you will define `Gates` that define access control.
+A `Gate` can then be used to create a `GatedAction` or a `GatedFunc`.
+Those are istances of the `GatedCode` class. `GatedCode` can be invoked in a `GateContext`.
+To create a `GateContext` you will need a `GatedRequest` containing user information
+and an instance of `IMachineInformation` for current machine environment information.
+
+Usually it will look like this:
+
+1. Service receives a user request
+2. A `GatedRequest` is constructed for the user request
+3. A `GateContext` is constructed with the `GatedRequest`
+4. A gated code path is accessed
+5. The `GatedCode` is invoked in the `GateContext`
+6. `GateContext` decides if `Gate` conditions are satisfied
+
+_From the [Gating.Example](https://github.com/microsoft/Omex/tree/master/src/Gating.Example):_
+
+    Gates gates = new Gates(GateDataSetLoader);
+
+    IGatedRequest gatedRequest = new SampleGatedRequest();
+    IGateContext gateContext = new GateContext(gatedRequest, machineInformation, null);
+    gateContext.PerformAction(
+        new GatedAction(
+            gates.GetGate("sample_allowed_gate"),
+            () => Console.WriteLine("SampleAction has been called")));
+
+## Usage scenarios
+
+For each gate defined in the XML configuration you'd create a class containing
+the appropriate definitions. This can be done using the `gategen` tool.
```

Comment:
You can add a link here to the CodeGenerators/GateGen project that I just merged (link to dir or to the readme file maybe).